src/misc/lv_anim.c

```diff
@@ -519,8 +546,13 @@ static void anim_timer(lv_timer_t * param)
 
     while(a != NULL) {
         uint32_t elaps = lv_tick_elaps(a->last_timer_run);
-        a->act_time += elaps;
 
+        if(a->is_paused) {
+            a->is_paused = (lv_tick_get() - a->pause_time) < a->pause_duration;
```

Comment:
Would this be applicable?

```suggestion
            uint32_t time_spent_paused = lv_tick_get() - a->pause_time;
            if(time_spent_paused >= a->pause_duration) {
                a->is_paused = false;
                uint32_t pause_overrun = time_spent_paused - a->pause_duration;
                a->act_time += pause_overrun;
            }
```

Reply:
Also, maybe instead of `lv_tick_get() - a->pause_time`, `lv_tick_elaps(a->pause_time)` could be used to handle tick overflow wrap-around as used by `elaps`.

Reply:
Wouldn't updating the `act_time` mean a small "jump" when the animation is resumed ? It would probably go unnoticed as long as the `pause_overrun` is not too big but it got me wondering

Reply:
There would be a jump, but there would be an even bigger jump at the next refresh without it. Here's a test case that passes with the change and fails without

```c
void test_anim_pause_for(void)
{
    int32_t var;

    /*Start an animation*/
    lv_anim_t a;
    lv_anim_init(&a);
    lv_anim_set_var(&a, &var);
    lv_anim_set_values(&a, 0, 100);
    lv_anim_set_exec_cb(&a, exec_cb);
    lv_anim_set_duration(&a, 100);
    lv_anim_t * animation = lv_anim_start(&a);

    lv_anim_pause_for(animation, 20);

    lv_test_wait(40);
    TEST_ASSERT_EQUAL(19, var);
}
```

Reply:
Also this more important issue. The overrun is lost and not counted. In fairness, it would only be noticeable if the refresh rate was very low.

```c
    ...
    lv_anim_t * animation = lv_anim_start(&a);

    lv_anim_pause_for(animation, 20);

    lv_test_wait(40);
    lv_test_wait(40);
    TEST_ASSERT_EQUAL(59, var); // "Expected 59 Was 39"
}
```

Reply:
Thank you for providing these test cases and i agree with you, will add the test cases and update the code. Nice catch !